sycl/include/sycl/known_identity.hpp

```diff
@@ -258,16 +258,13 @@ template <typename BinaryOperation, typename AccumulatorT>
 struct known_identity_impl<BinaryOperation, AccumulatorT,
                            std::enable_if_t<IsMinimumIdentityOp<
                                AccumulatorT, BinaryOperation>::value>> {
-// TODO: detect -fno-honor-infinities instead of -ffast-math
-// See https://github.com/intel/llvm/issues/13813
-// This workaround is a vast improvement, but still falls short.
-// To correct it properly, we need to detect -fno-honor-infinities usage,
-// perhaps via a driver inserted macro.
-// See similar below in known_identity_impl<IsMaximumIdentityOp>
-#ifdef __FAST_MATH__
-  // -ffast-math implies -fno-honor-infinities,
-  // but neither affect ::has_infinity (which is correct behavior, if
-  // unexpected)
+
+#if defined(__FINITE_MATH_ONLY__) && (__FINITE_MATH_ONLY__ == 1)
+  // Finite math only (-ffast-math,  -fno-honor-infinities) improves
+  // performance, but does not affect ::has_infinity (which is correct behavior,
+  // if unexpected). Use ::max() instead of ::infinity().
+  // Note that if someone uses -fno-honor-infinities WITHOUT -fno-honor-nans
+  // they'll end up using ::infinity(). This is a very unlikely occurence
```

Comment:
> This is a very unlikely occurence

This needs to be explained.

Reply:
It was felt by the FE team that there is no reasonable case where one of the flags would be used without the other and therefore `__FINITE_MATH_ONLY__` is sufficient for this problem ( and superior to the `__FAST_MATH__` macro we were using before).

Reply:
It needs to be explained in the code for future readers, not in the code-review.